### ISSUE TITLE
Parameters for /collections/{id}?

### ISSUE THREAD

**Issue by m-mohr** [opengeospatial/ogcapi-common#165]
This [document](https://github.com/opengeospatial/oapi_common/blob/August-2020/collections/clause_8_collections.adoc)
says:

> Three parameters are also defined for use in these operations.

"These operations" seem to be /collections, /collections/{collectionId} and /collections/{collectionId}/items from table 1. While these parameters (table 2) make sense for the first and last endpoint, it doesn't seem very useful for the second. Should this be clarified?

**Comment by cmheazel** [comment]
The /collections/{collectionId} GET operation is defined in section 8.1.2. This section states in the Parameters section that "No parameters have been standardized for this operation."

**Comment by m-mohr** [comment]
True, but I'm not speaking about 8.1.2. The chapter 8 I'm referring to is still not precise on its own. The sentence should mention the operation names explicitly.

**Comment by cmheazel** [comment]
Notes to self:
None of these parameters are mandatory.  Their use is up to the developer of the API (section 8.2).
The requirements are not faithful to this assertion. They need to be fixed.
The behavior of these parameters is only defined for two of the three operations. 
Even then, the applicability of bbox and datetime to /items is debatable and should probably be removed.
Actions:
Make clear that these parameters are re-usable modules
Their behavior (if applicable) should be individually defined for each resource.
Determine what a server should return to a client which uses a parameter that is not supported by the server but who's behavior is defined in the standard - keeping in mind that these parameters are optional so we are looking for a soft failure.
General issue:
Not all modules are significant enough to merit their own conformance class.  Can we define behaviors in such a way that failure to support one of these optional modules does not impact interoperability?

**Comment by cmheazel** [comment]
PR 271 updated March 17:
1) The applicable parameters are identified for each resource/operation pair
2) All parameters are optional
3) The encoding of each parameter is specified in section 8.2
4) General requirements for processing each parameter are provided in section 8.2
5) Specific requirements for processing each applicable parameter are provided for each resource/operation pair in section 8.1
6) Requirements for the server response when an applicable parameter is not supported by that server are provided in the resource/operation sections of section 8.1

In summary:
1) A client may use any parameter documented in the standard
2) A server may:
    a) Provide support for a parameter as documented in the standard
    b) Not provide support for a parameter as long as the server handles the parameter in conformance with the "unsupported" requirements for that parameter.

The result should be maximum flexibility for the server developers and minimal discomfort for clients.

Note: this conflicts with requirement 26 (query-param-unknown) which is also included in Part 1.

**Comment by cmheazel** [comment]
Modified requirement 26 (query-param-unknown) to allow a non-400 response if /per/core/query-param-specified or /per/core/query-param-tolerance are applicable. This fixes the requirement conflict identified above.

**Comment by cmheazel** [comment]
Re-wrote prior paragraph.  In API-Common Part 2 these operations are only applicable to the /collections resource.  The text has been updated accordingly.

**Comment by cmheazel** [comment]
June 26, 2021 - close NOTUC

**Comment by m-mohr** [comment]
Out of curiosity: What is NOTUC?

**Comment by joanma747** [comment]
No Objection To Unanimous Consent (among the ones present in the meeting)
